src/lib.rs

```diff
@@ -175,14 +177,26 @@ impl Form {
                 }
             }
         }
-        Ok(Form { doc, form_ids })
+        Ok(Form { document, form_ids })
     }
 
     /// Returns the number of fields the form has
     pub fn len(&self) -> usize {
         self.form_ids.len()
     }
 
+    /// Gets the Acroform object id
+    pub fn get_form(&self) -> Result<ObjectId, lopdf::Error> {
```

Comment:
This seems like a leaky abstraction to me.  Ideally, any functionality that we want to expose should handle the lopdf stuff internally.  Why do we want this?

Reply:
Yes, it's true but think this: you want to add a field inside the Form, what you do? Save the file then reload it with lopdf?

Reply:
I don't think exposing the internals counts as "supporting" the ability to add fields.  It allows you to add fields, but I wouldn't call it "support".  I think that either we don't support it and don't have this method or we do support it and we have dedicated, non-leaky methods to add fields.

Reply:
Mmmmh I think you are right, I removed this method, maybe in the future we can add the ability to add fields but at least, we have to build a struct for different field format that we add abstractly into the lopdf Document.

Anyway, I think it's correct to expose the `document` property and then the `get_object_id()` function will let you to manipulate directly a field or get the page of a field using the `get_object_page()` of lopdf.

I can explain why this is needed: you want to add an Image beside a field, you need the page of a field and for get the page you need the `document` and then with `get_object_page()` you obtain the `ObjectId` of a page and then, manipulating the `document`, you can insert an image or whatever, without the document you can't access the field and you have to reprocess the document externally again, it's very boring, it will very helpful to other if the library exposes the document field imho.

Also, I added a `remove_field()` method in order to be less leaky abstraction :-)

Please check now :-)

Reply:
@jsandler18 any chance to check this? :-)

Reply:
I can see the use in being able to modify the PDF itself, but I don't think it should be added haphazardly when we think of it.  Right now, you can only read and update the field values.  If we want to add the ability to mutate the structure of the form, I think we should think about what we want exactly and add that all at once, rather than exposing our internals and saying "you figure it out".

Reply:
@jsandler18 applied your suggestions, yes, in this case... we have to think on it a bit deeper so in the meantime, I just updated the **lopdf** library and added the `remove_field()` method.

Reply:
Why are we keeping remove_field?

Reply:
As I think is a related function, the library should be ready for remove fields, what you think?

Reply:
I don't think we should include it right now for the reason I already stated: the ability to mutate the structure of the form itself should be more thought out, and separate from this pr

Reply:
I don't think so, this PR is ONLY for this new method, we can change their scope as if you look on the changes now, it adds only this method, I can change the description with "added `remove_field()` method and updated `lopdf`"

Reply:
Sorry for the delay, I have been busy.  I see that the only substantial change here is adding the remove_field method, but I am not sure I see the point of it.

Reply:
Personally I'm using it but as I need some other features too, I'm using the form of mine, maybe a day I'll do something better for fields handling, removing and better structure instead of an index or name. You can close this PR if you prefer :-)